src/lib/y2network/sysconfig/dns_reader.rb

```diff
@@ -75,7 +79,15 @@ def resolv_conf_policy
       #
       # @return [String]
       def hostname
-        HostnameReader.new.hostname
+        @hostname_reader = HostnameReader.new
+        @hostname_reader.hostname
+      end
+
+      # Checks whether the hostname was read from install.inf
+      #
+      # @return [Boolean]
+      def hostname_from_install_inf?
+        !@hostname_reader.install_inf_hostname.nil?
```

Comment:
this is really nasty side effect that really depends on fact that hostname entry is constructed before save_hostname entry. What I really prefer is method `hostname_reader` that returns cached object and those two methods just call it.

Reply:
I've follow up of this PR. I separate Hostname and DNS classes there exactly for this (and one above) reason. So, would you prefer to merge those two patches together?

Reply:
I am fine with merging it together, so I can see final result.